src/NuGet.Core/NuGet.Frameworks/FallbackFramework.cs

```diff
@@ -54,7 +60,8 @@ public bool Equals(FallbackFramework other)
             }
 
             return NuGetFramework.Comparer.Equals(this, other)
-                && NuGetFramework.Comparer.Equals(Fallback, other.Fallback);
+                && Fallback.All(p => other.Fallback.Contains(p, NuGetFramework.Comparer))
```

Comment:
Use SequenceEqual here. In the Hash above the order matters. I would say the order should matter here also since the fallback is ordered.